lib/logstash/inputs/udp.rb

```diff
@@ -39,34 +39,17 @@ def initialize(params)
 
   public
   def register
-    @udp = nil
-  end # def register
-
-  public
-  def run(output_queue)
-  @output_queue = output_queue
-    begin
-      # udp server
-      udp_listener(output_queue)
-    rescue => e
-      @logger.warn("UDP listener died", :exception => e, :backtrace => e.backtrace)
-      Stud.stoppable_sleep(5) { stop? }
-      retry unless stop?
-    end # begin
-  end # def run
-
-  private
-  def udp_listener(output_queue)
     @logger.info("Starting UDP listener", :address => "#{@host}:#{@port}")
 
-    if @udp && ! @udp.closed?
-      @udp.close
-    end
-
     @udp = UDPSocket.new(Socket::AF_INET)
```

Comment:
I think I was just concerned about the change of behaviour, now the plugin reconnect itself but with this changes is not going to do it anymore. I understand the semantic implications and I agree, although I'm not sure we should do this change like this, might be we should also improve the restart semantic in the pipeline before getting into this plugin changes. What do you think?